TextInputKit/TextInputKit/Code/TextInputFormatter/TextInputValidationResult+OptimizeChanges.swift

```diff
@@ -12,9 +12,19 @@ public extension TextInputValidationResult {
         forEditing originalString: String,
         replacing replacementString: String,
         at editedRange: Range<String.Index>,
+        withSelection originalSelectedRange: Range<String.Index>,
         resulting resultingString: String,
         withSelection resultingSelectedRange: Range<String.Index>) -> TextInputValidationResult {
 
+        let shouldReject: Bool = {
```

Comment:
Yes.

So, if we imagine a pipeline of text input events, we can prevent calls to the event handlers in advance.